digilan-token.php

```diff
@@ -1141,8 +1225,44 @@ public static function authenticate_ap_user_on_wp($username = '')
         }
         error_log($social_id . ' has logged in with ' . $provider);
         $user_id = DigilanTokenUser::select_user_id($mac, $social_id);
+
+        $user_info = array();
         if ($user_id == false) {
-            DigilanTokenUser::create_ap_user($mac, $social_id);
+            foreach ($_GET as $get_key -> $get_value) {
+                [$prefix, $field_key] = explode('/', $get_key);
+                if ($prefix !== 'custom-form-portal-hidden') {
+                    continue;
+                }
+
+                $field_value;
+                switch ($field_key) {
+                    case 'text':
+                        $field_value = DigilanTokenSanitize::sanitize_custom_form_portal_hidden_text($get_value);
+                        break;
+                    case 'number':
+                        $field_value = DigilanTokenSanitize::sanitize_custom_form_portal_hidden_number($get_value);
+                        break;
+                    case 'email':
+                        $field_value = DigilanTokenSanitize::sanitize_custom_form_portal_hidden_email($get_value);
+                        break;
+                    case 'tel':
+                        $field_value = DigilanTokenSanitize::sanitize_custom_form_portal_hidden_tel($get_value);
+                        break;
```

Comment:
for each inconu do if ( somethingX )  do somethingY
vers
try_do_somethingY(somethingX )

merci ( pas de for pour faire jolie, des fonctions pour faire claire )